recipe/meta.yaml

```
@@ -344,16 +345,18 @@ outputs:
   {% set pytorch_cpu_gpu = "pytorch-gpu" %}   # [cuda_compiler_version != "None"]
   - name: {{ pytorch_cpu_gpu }}
     build:
-      string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
-      string: cpu_{{ blas_impl }}_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                      # [cuda_compiler_version == "None"]
+      string: cuda{{ cuda_compiler_version | replace('.', '') }}_{{ blas_impl }}py{{ CONDA_PY }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
+      string: cpu_{{ blas_impl }}_py{{ CONDA_PY }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                                # [cuda_compiler_version == "None"]
```

pytorch-gpu should not have `py{{ CONDA_PY }}` for megabuilds, compare main
https://github.com/conda-forge/pytorch-cpu-feedstock/blob/f3f31a9ee57ab64d53cbc171502ed477e3ac76df/recipe/meta.yaml#L517-L520

AFAICT the third line there is never used (because we don't have non-megabuilds -- i.e. only osx -- together with CUDA). That also means the missing `_` between `{{ blas_impl }}` and `py{{ CONDA_PY }}` does not actually make it into any published artefacts; Based on existing builds, we should definitely keep that underscore. Something like
```suggestion
      string: cuda{{ cuda_compiler_version | replace('.', '') }}_{{ blas_impl }}_h{{ PKG_HASH }}_{{ build }}                  # [megabuild and cuda_compiler_version != "None"]
      string: cpu_{{ blas_impl }}_h{{ PKG_HASH }}_{{ build }}                                                                 # [megabuild and cuda_compiler_version == "None"]
      # only applicable to osx, where we don't have CUDA builds
      string: cpu_{{ blas_impl }}_py{{ CONDA_PY }}_h{{ PKG_HASH }}_{{ build }}                                                # [not megabuild]
```

---

I figured there must be a reason why there's no `megabuild` logic there, but if you say it's wrong, I've just copied the remaining changes from `main` (only with `build` changed to `PKG_BUILDNUM` for consistency). Does that address the concerns?

---

Copying from main is the right move, aside from the abovementioned fact that `# [not megabuild and cuda_compiler_version != "None"]` is pointless (which is why I made a concrete suggestion). But fine by me to leave as it is now for this PR.